### PR TITLE
Refactor `aria-describedby` on form groups

### PR DESCRIPTION
**Pull request by alex-ju**
### :pushpin: Summary

Refactor `aria-describedby` on form groups so that helper text and error messages at group level are available to screen reader users when landing on a form control within the group.

### :hammer_and_wrench: Detailed description

We switch from associating group level helper text and error messages with the fieldset to associating them with individual input items within the group as it seem to have better screen reader support.

**Note**
Unfortunately, the use of `concat` helper means end up with some leading or trailing whitespaces inside the `aria-describedby` which I'm not sure how to avoid.

### :link: External links

[Jira](https://hashicorp.atlassian.net/jira/software/c/projects/HDS/boards/445?modal=detail&selectedIssue=HDS-261) [issues](https://hashicorp.atlassian.net/jira/software/c/projects/HDS/boards/445?modal=detail&selectedIssue=HDS-262)

***

### 👀 How to review

👉 Review by files changed

Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

**Review by MelSumner — APPROVED** [review]
I've tested this with a few combinations and the help and error text are being associated correctly. If I find any bugs when I test on other devices/combinations, I'll file a separate bug; for now I have enough confidence that this will work as designed.

File: packages/components/addon/components/hds/form/utils/setAriaDescribedBy.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export function getAriaDescribedBy(element) {
+export function setAriaDescribedBy(element) {
```
Comment (didoo) [inline packages/components/addon/components/hds/form/utils/setAriaDescribedBy.js:1]: I have a couple of alternatives that may simplify the code further (`element.descriptors` is already an array, so no need to re-build it via a `forEach` and `push` on a new array).

```
export function setAriaDescribedBy(element) {
  let ariaDescribedBy = element.descriptors || [];

  // append @extraAriaDescribedBy arg, if provided
  if (element.args.extraAriaDescribedBy) {
    element.descriptors.push(element.args.extraAriaDescribedBy);
  }

  if (ariaDescribedBy.length) {
    element.ariaDescribedBy = ariaDescribedBy.join(' ');
  }
}
```

or if you want to completely get rid of the temp local variable `ariaDescribedBy` you could use directly the `element.descriptors` array itself:

```
export function setAriaDescribedBy(element) {
  // append @extraAriaDescribedBy arg, if provided
  if (element.args.extraAriaDescribedBy) {
    element.descriptors.push(element.args.extraAriaDescribedBy);
  }

  if (element.descriptors.length) {
    element.ariaDescribedBy = element.descriptors.join(' ');
  }
}
```

**Review by didoo — APPROVED** [review]
I've left a couple of suggestions (not blockers).

I like this final implementation, definitely much cleaner (and easy to understand) than the previous one.

In any case, good to go for me.

File: packages/components/addon/components/hds/form/field/index.js [hunk 2]
```diff
@@ -3,18 +3,26 @@ import { tracked } from '@glimmer/tracking';
...
+import { setAriaDescribedBy } from '../utils/setAriaDescribedBy';
+import { schedule } from '@ember/runloop';
 
 export const LAYOUT_TYPES = ['vertical', 'flag'];
 
 export default class HdsFormFieldIndexComponent extends Component {
-  @tracked ariaDescribedBy = getAriaDescribedBy(this);
-  @tracked descriptors = [];
+  @tracked ariaDescribedBy = this.args.extraAriaDescribedBy;
+  @tracked descriptors = this.args.descriptors || [];
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply packages/components/addon/components/hds/form/field/index.js:13]: good spot! – leftover from trying different approaches

File: packages/components/addon/components/hds/form/utils/setAriaDescribedBy.js (hunk 1 above)
Comment (alex-ju) [reply packages/components/addon/components/hds/form/utils/setAriaDescribedBy.js:1]: Indeed, now that we fixed the async issue that made me keep that counterintuitive iteration, we can simply this code further and I like your suggestion above

**Review by MelSumner — APPROVED** [review]
Still have no objections to this approach. 👍

**Pull request merged**